src/character.cpp

```diff
@@ -596,6 +596,54 @@ std::vector<item_location> Character::nearby( const std::function<bool(const ite
     return res;
 }
 
+long int Character::i_add_to_container(const item &it)
```

Comment:
Instead of using `u.inv` and `u.worn` explicitly, you could use `visitable<T>::visit_items`. Then it would be easy to also apply on drop on map/vehicle (those are also `visitable`, so that you could reload a stack of quivers in the trunk by dropping a stack of arrows on the trunk.

Using `visitable` would not change the preference to reload worn items first - for characters, it visits items in order: `weapon->worn->inv`.